src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  // create copy to protect input data from changes
+  const stateCopy = { ...state };
+  const result = [];
+
+  for (const item of actions) {
+    switch (item.type) {
+      case 'addProperties':
+        for (const key in item.extraData) {
+          stateCopy[key] = item.extraData[key];
+        };
+        // пушимо через деструктурізацію, зберігаючи модифікований об'єкт
+        result.push({ ...stateCopy });
```

Comment:
Here and in the next cases you repeat the same line ```result.push({ ...stateCopy }); ``` - you can do it after case block. Checklist 4. [DONT REPEAT YOURSELF] - If you perform same action in all `switch` cases - do it just once afterwards.